api/v1alpha1/tykoasapidefinition_types.go

```diff
@@ -75,6 +111,14 @@ type TykOasApiDefinitionStatus struct {
 	LatestTransaction TransactionInfo `json:"latestTransaction,omitempty"`
 	// IngressTemplate shows whether this CR is used as Ingress Template or not.
 	IngressTemplate bool `json:"ingressTemplate,omitempty"`
+	// IsVersionedAPI indicates if the API is a version or not.
+	IsVersionedAPI bool `json:"isVersionedAPI,omitempty"`
+	// BaseVersionName specifies the name of the Base OAS API if IsVersionedAPI is set to true.
+	BaseVersionName string `json:"baseVersionName,omitempty"`
+	// BaseVersionNamespace specifies the namespace of the Base OAS API.
+	BaseVersionNamespace string `json:"baseVersionNamespace,omitempty"`
+	// IsDefaultVersion specifies if the OAS API is the default  Version.
+	IsDefaultVersion bool `json:"isDefaultVersion,omitempty"`
```

Comment:
i would suggest moving all versioning-related info to a new sub-structure maybe called as `*VersioningStatus`. And set it to nil if the OAS API is not versioned.